platform-ajax/src/main/java/com/softicar/platform/ajax/testing/selenium/grid/SleniumGridDockerComposeFileGenerator.java

```diff
@@ -91,23 +90,21 @@ private void appendSeleniumHubService(OutputBuilder builder) {
 
 	private void appendNodeServices(OutputBuilder builder) {
 
-		for (ISeleniumNodeConfiguration nodeConfiguration: nodeConfigurations) {
-			for (int nodeIndex = 1; nodeIndex <= calculateNodeCount(nodeConfiguration); nodeIndex++) {
-				builder.appendLine("  %s:", generateNodeName(nodeConfiguration, nodeIndex));
-				builder.appendLine("    image: selenium/%s:%s", nodeConfiguration.getImageName(), gridConfiguration.getContainerVersion());
-				builder.appendLine("    container_name: %s", generateNodeName(nodeConfiguration, nodeIndex));
-				builder.appendLine("    volumes:");
-				builder.appendLine("      - /dev/shm:/dev/shm");
-				builder.appendLine("    depends_on:");
-				builder.appendLine("      - %s", generateHubName());
-				builder.appendLine("    environment:");
-				builder.appendLine("      - SE_EVENT_BUS_HOST=%s", generateHubName());
-				builder.appendLine("      - SE_EVENT_BUS_PUBLISH_PORT=%s", hubConfiguration.getPortEventBusPublish());
-				builder.appendLine("      - SE_EVENT_BUS_SUBSCRIBE_PORT=%s", hubConfiguration.getPortEventBusSubscribe());
-				builder.appendLine("    networks:");
-				builder.appendLine("      - %s", generateNetworkName());
-				builder.appendLine("");
-			}
+		for (int nodeIndex = 1; nodeIndex <= calculateNodeCount(nodeConfiguration); nodeIndex++) {
```

Comment:
should be extracted to variable
```suggestion
		var nodeCount =  calculateNodeCount(nodeConfiguration);
		for (int nodeIndex = 1; nodeIndex <= nodeCount; nodeIndex++) {
```